### PR TITLE
Added a Scheduling class to schedule one-off and recurring events

### PR DESCRIPTION
**Pull request by Leterax**
A simple wrapper for the standard libs `sched` library to allow scheduling events

### CODE REVIEW

File: tests/test_scheduler.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import time
+from unittest import TestCase
+from moderngl_window.utils.schedule import Scheduler
```
Comment (einarf) [inline tests/test_scheduler.py:3]: Wrong import. The module is called `scheduler`

File: moderngl_window/utils/scheduler.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+
+class Scheduler:
+    def __init__(self, timer=time.time):
+        """Create a Scheduler object to handle events.
+
+        :param timer: timer class or time function to use, defaults to time.time.
+        :type timer: callable, optional
+        :raises ValueError: when timer is not a valid argument
+        """
```
Comment (einarf) [inline moderngl_window/utils/scheduler.py:13]: We use google style docstrings.

```
Args:
    name (type): Description

Keyword Args:
    name (type): Description
```

File: moderngl_window/utils/scheduler.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        :param delay: delay in seconds
+        :type delay: float
+        :param priority: priority for this event, lower is more important, defaults to 1
+        :type priority: int, optional
+        :param arguments: arguments for the action, defaults to ()
+        :type arguments: tuple, optional
+        :param kwargs: keyword arguments for the action, defaults to dict()
+        :type kwargs: dict, optional
+        :return: returns a event id that can be canceled
+        :rtype: int
```
Comment (einarf) [inline moderngl_window/utils/scheduler.py:45]: We use google style docstrings.

```
Args:
    name (type): Description

Keyword Args:
    name (type): Description

Returns:
   type
```

File: moderngl_window/utils/scheduler.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        :param time: epochtime in seconds
+        :type time: float
+        :param priority: priority for this event, lower is more important, defaults to 1
+        :type priority: int, optional
+        :param arguments: arguments for the action, defaults to ()
+        :type arguments: tuple, optional
+        :param kwargs: keyword arguments for the action, defaults to dict()
+        :type kwargs: dict, optional
+        :return: returns a event id that can be canceled
+        :rtype: int
```
Comment (einarf) [inline moderngl_window/utils/scheduler.py:66]: We use google style docstrings.

```
Args:
    name (type): Description

Keyword Args:
    name (type): Description

Returns:
   type
```

File: moderngl_window/utils/scheduler.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        :param delay: delay in seconds
+        :type delay: float
+        :param priority: priority for this event, lower is more important, defaults to 1
+        :type priority: int, optional
+        :param arguments: arguments for the action, defaults to ()
+        :type arguments: tuple, optional
+        :param kwargs: keyword arguments for the action, defaults to dict()
+        :type kwargs: dict, optional
+        :return: returns a event id that can be canceled
+        :rtype: int
```
Comment (einarf) [inline moderngl_window/utils/scheduler.py:89]: We use google style docstrings.

```
Args:
    name (type): Description

Keyword Args:
    name (type): Description

Returns:
   type
```

File: moderngl_window/utils/scheduler.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        :param arguments: arguments for the function
+        :type arguments: tuple
+        :param kwargs: keyword arguments for the function
+        :type kwargs: dict
+        :param scheduling_info: tuple of information for scheduling the event
+        :type scheduling_info: tuple
+        :param id: the id the event will get
+        :type id: int
+        :return: returns a callable that will reschedule itself
+        :rtype: callable
```
Comment (einarf) [inline moderngl_window/utils/scheduler.py:115]: We use google style docstrings.

```
Args:
    name (type): Description

Keyword Args:
    name (type): Description

Returns:
   type
```

**Comment by Leterax** [comment]
I don't like the way the Scheduler is being created. Passing a timer class or a time function just seems weird and the very hacky checks inside init don't feel right. I'm not sure how to handle it better, since sometimes a timer class isn't there and you infact have to pass your own time function but it still feels weird.

**Pull request merged**